image-sync-formula/_modules/image_sync.py

```diff
@@ -67,11 +67,17 @@ def filter_images(whitelist = [], arch_list = ['x86_64', 'i586', 'i686']):
 def deleted_images(arch_list = ['x86_64', 'i586', 'i686']):
     pillar_images = __pillar__.get('images', {})
     grains_images = __grains__.get('images', {})
+    grains_boot_images = __grains__.get('boot_images', {})
+    grains_pxe_entries = __grains__.get('pxe_entries', {}).values()
     res = []
     for image_id, image_versions in grains_images.items():
         for image_version, image_data in image_versions.items():
             if image_data.get('arch') not in arch_list:
                 continue
+            if (image_data.get('boot_image') in grains_pxe_entries
+               and grains_boot_images.get(image_data.get('boot_image'), {}).get('sync', {}).get('kernel_link')):
+                # this is a bundle and the bundled boot image is still in use in a pxe entry
+                continue
```

Comment:
Why do we need to keep bundle image and not just extracted boot image?

In `deleted_boot_images` we get the list of `deleted_bundle_boot_images` which if this was not there indeed would contain our boot image, but before that check we check for `boot_image_id` in `grains_pxe_entries` which would hit and continue.

What am I missing?

Reply:
For bundles, the boot image directory contains just symlinks to the bundle image directory.

We could work with individual files in the bundle directory, but I don't think it is worth the effort, when the bundles are not used by default.